vfs/read.go

```diff
@@ -62,7 +65,11 @@ func (fh *ReadFileHandle) openPending() (err error) {
 	if fh.opened {
 		return nil
 	}
-	r, err := fh.o.Open()
+	var end int64 = -1
+	if fh.chunkSize > 0 {
+		end = fh.chunkSize - 1
```

Comment:
This should probably cap end to o.Size(), or maybe add another condition in the if fh.chunkSize > 0 && fh.chunkSize < o.Size()?

Reply:
The check should not be necessary. The HTTP RFC [https://tools.ietf.org/html/rfc7233#section-2.1](7233#2.1) specifies: 
>A client can limit the number of bytes requested without knowing the
> size of the selected representation.  If the last-byte-pos value is
> absent, or if the value is greater than or equal to the current
> length of the representation data, the byte range is interpreted as
> the remainder of the representation (i.e., the server replaces the
> value of last-byte-pos with a value that is one less than the current
> length of the selected representation).

In short, the HTTP server has to correct the end value to the correct position.
My tests using Google Drive confirmed this behavior.

I can add the check to reduce the chance of a broken HTTP server rejecting requests beyond the real end.

Reply:
I think RangeOption should follow the RFC here, so your code is good as it stands.  I added some extra docs to the help in 84bc4dc142424ad0bc9f9896939e5a7c09078a5c and as part of #1825 I'll make sure this gets integration tested.